src/App.jsx

```diff
@@ -1,19 +1,29 @@
-import { useState } from 'react'
+import { useState, useEffect } from 'react'
 import reactLogo from './assets/react.svg'
 import viteLogo from '/vite.svg'
 import './App.css'
 import AddTodoForm from './AddTodoForm'
 import TodoList from './TodoList'
 
+function useSemiPersistentState() {
+  const [todoList, setTodoList] = useState(JSON.parse(localStorage.getItem('savedTodoList')));
```

Comment:
When you run this line in an environment that doesn't have set the local storage, the app will crash, because the `localStorage.getItem('savedTodoList')` will be null, and set the todoList with null as a initial value, so you should add a short circuit in case this is null, like:
`JSON.parse(localStorage.getItem('savedTodoList')) || []`
So the first time if JSON.parse(localStorage.getItem('savedTodoList')) is null, it'll be considered as false and todoList will be set as []